lms/static/scripts/frontend_apps/utils/cfi.ts

```diff
@@ -0,0 +1,68 @@
+/**
```

Comment:
This code is copied from `cfi.ts` in the Hypothesis client repository, keeping only the used parts. Same for the tests.